osh/hub/other/middleware.py

```diff
@@ -0,0 +1,33 @@
+from django.http import HttpResponse
+from django.template.loader import get_template
+from django.urls import reverse
+
+
+class LoginRequiredMiddleware:
+    """
+    Specially hacked middleware to make sure the user list/detail page are
+    visible only to authenticated users.
+
+    This is an ugly hack based on kobo's implementation of `users`. Updating
+    kobo would be more straightforward, but that might break other modules
+    that depend on kobo.
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        path_info = request.META["PATH_INFO"]
+        # This `user/list` comes from kobo's view name for user
+        kobo_user_view_url = reverse("user/list")
+        if path_info.startswith(kobo_user_view_url):
+            if not request.user.is_authenticated:
+                template = get_template("base.html")
+                context = {
+                    "info_message": "Please login to view users.",
```

Comment:
Note: This allows to show a banner message in case of anonymous access. It's from `templates/layout.html` in kobo.